### PR TITLE
Use doxygen to document the resource manager code.

### PR DESCRIPTION
**Pull request by ifakhrutdinov**

### CODE REVIEW

File: h/resmgr.h [hunk 1]
```diff
@@ -40,69 +54,57 @@ ZOWE_PRAGMA_PACK_RESET
...
-*   handle                    - handle containing resource manager information
-*   managerRC                 - resource manager return code
-*
-* Return value:
-*   RC_RESMGR_OK is returned when resource manager has been added.
-*   RC_RESMGR_SERVICE_FAILED is returned when an error occurred (see managerRC
-*   for details).
-*****************************************************************************/
+/**
+ * @brief Adds a task resource manager.
```
Comment (daveyc) [inline h/resmgr.h:58]: Should we code @brief commands? Doxygen can be configured to implicitly handle brief comments using the JAVADOC_AUTOBRIEF option.

Comment (ifakhrutdinov) [reply h/resmgr.h:58]: Not sure. I've looked at existing projects and many use `@brief` explicitly.

Comment (daveyc) [reply h/resmgr.h:58]: I tend to stick to Java compatible doc comments. But it's not a big issue for me.

**Review by daveyc — APPROVED** [review]
LGTM

**Review by 1000TurquoisePogs — APPROVED** [review]
Nice, thanks!

**Pull request merged**